R/SEIRmodels.R

```diff
@@ -0,0 +1,749 @@
+                              "IMVGetLive1", "IMVGetLive2", "IMVNotGetLive1",
+                              "IMVNotGetLive2", "IMVGetDie1", "IMVGetDie2",
+                              "IMVNotGetDie1", "IMVNotGetDie2", "IRec1",
+                              "IRec2", "R", "D"))
+
+  #df <- tidyr::pivot_longer(tibble::as_tibble(df), cols = df$nameslist)
+
+  strname <- paste("SIR", df$type, "Model Simulation")
+
+  ggplot2::ggplot(
+                  df, ggplot2::aes(x = df$time, y = df$value,
+                  group = interaction(df$group, df$name), colour = df$name)) +
+    ggplot2::geom_line(size = 0.5) +
+    ggplot2::theme_bw() +
+    ggplot2::labs(title = strname, subtitle = subtitle, color = df$legend) +
+    ggplot2::labs(y = "States", x = "time") +
+    ggplot2::theme(
+      legend.justification = c("right", "top"),
+      legend.box = c("horizontal", "vertical")
+    ) +
+    ggplot2::theme(
+      text = ggplot2::element_text(color = "#444444", family = "Lucida Bright"),
+      plot.title = ggplot2::element_text(size = 26, color = "#333333"),
+      plot.subtitle = ggplot2::element_text(size = 13),
+      axis.title.x = ggplot2::element_text(size = 16, color = "#333333"),
+      axis.title.y = ggplot2::element_text(angle = 0, vjust = .5))
+
+}
+
+
+#' @title Infection_process -> S to I
+#'
+#' @param S susceptible
+#' @param I infected
+#' @param human human
+#' @param immunity immunity
+#' @param age age
+#' @param location location
+#' @param pars parameter list
+#'
+#' @export
+#'
+#' @examples
+#' Individual_S_to_I(S, I, immunity, age, location, pars)
+#' @importFrom stats runif
+Individual_S_to_I <- function(
+  S, I, human, immunity, age, location, pars = NULL) {
+  function(api) {
+
+    pars <- Get_parameters_for_sirstochastic(pars)
+
+    # calculate information for infections, recoveries and births
+    inf <- Infections(length(api$get_state(human, I)),
+                      length(api$get_state(human, S)), pars)
+
+    n_to_infect <- inf$n_infections_S
+    susceptible <- api$get_state(human, S)
+
+    if (pars$novariations) {
+      infected <- susceptible[sample.int(length(susceptible), n_to_infect)]
+      api$queue_state_update(human, I, infected)
+    }
+    if (pars$includeimmune) {
+      # Get the immunity for susceptible humans and use the complement to modify
+      # the infection rate
+      rate_modifier <- 1 - api$get_variable(human, immunity, susceptible)
+      prob <- pars$infection_rate * rate_modifier
+      individual::fixed_probability_state_change_process(
+        "human", S$name, I$name, prob)
+    }
+    if (pars$includeage) {
+      # Get the age for susceptible humans and use the complement to modify the
+      # infection rate
+      rate_modifier <- 1 - api$get_variable(human, age, susceptible)
+      prob <- pars$age_rate * rate_modifier
+      individual::fixed_probability_state_change_process(
+        "human", S$name, I$name, prob)
```

Comment:
If we are returning before the end of the function, it's clearer (and more efficient) to write:

```suggestion
      return(individual::fixed_probability_state_change_process(
        "human", S$name, I$name, prob))
```